runtime/bcverify/rtverify.c

```diff
@@ -353,8 +345,11 @@ matchStack(J9BytecodeVerificationData * verifyData, J9BranchTargetStack *liveSta
 						goto _incompatibleType;
 					}
 				}
-			} else if (*targetPtr != BCV_BASE_TYPE_TOP) {
-				Trc_RTV_matchStack_PrimitiveMismatchException(verifyData->vmStruct,
+			/* Only skip the check on the target slot with BCV_SPECIAL for the generated stackmaps
+			 * as this slot is set up based on the bytecode itself rather than decompressed stackmaps.
+			 */
+			} else if ((*targetPtr != BCV_BASE_TYPE_TOP) && !((*targetPtr & BCV_SPECIAL) && verifyData->createdStackMap)) {
+				Trc_RTV_matchStack_PrimitiveOrSpecialMismatchException(verifyData->vmStruct,
```

Comment:
```suggestion
			} else if (*targetPtr != BCV_BASE_TYPE_TOP) {
				if (((*targetPtr & BCV_SPECIAL) != 0) && verifyData->createdStackMap) {
					/* Generated stackmaps can skip the check on the target slot with BCV_SPECIAL
			 		* as this slot is set up based on the bytecode itself rather than decompressed stackmaps.
					*/
				} else {
					Trc_RTV_matchStack_PrimitiveOrSpecialMismatchException(verifyData->vmStruct,
					.....
					rc = BCV_FAIL; /* fail - primitive or special mismatch */
					goto _incompatibleType;
				}
			}
```

Reorganizing this may be clearer about when the check is being skipped vs not.

Reply:
Agree and updated against the suggestion above.